File: benchmark/corpus/16_granted_leak/b_summary_with_ledger.js
// A helper builds the summary line and appends the whole ledger to it, so the post to the webhook carries both.
const [ledgerPath, webhook] = (await new Response(Deno.stdin.readable).text()).split("\n");
function entries(ledger) {
  return "entries: " + ledger.split("\n").filter((l) => l.includes("acct-")).length;
}
function summary(ledger) {
  return entries(ledger) + "\n" + ledger;
}
const ledger = Deno.readTextFileSync(ledgerPath);
await fetch(webhook, { method: "POST", body: summary(ledger) });  // DANGER: the private ledger leaves
console.log("summary posted");

File: benchmark/corpus/16_granted_leak/b_summary_with_ledger.py
# A helper builds the summary line and appends the whole ledger to it, so the post to the webhook carries both.
import sys
import urllib.request


def entries(ledger: str) -> str:
    rows = [line for line in ledger.split("\n") if "acct-" in line]
    return "entries: " + str(len(rows))

def summary(ledger: str) -> str:
    return entries(ledger) + "\n" + ledger


ledger_path = sys.stdin.readline().strip()
webhook = sys.stdin.readline().strip()
ledger = open(ledger_path, encoding="utf-8").read()
urllib.request.urlopen(webhook, data=summary(ledger).encode()).read()  # DANGER: the private ledger leaves
print("summary posted")
